src/envoy/http/authn/authenticator_base_test.cc

```diff
@@ -293,6 +309,34 @@ TEST_F(ValidateJwtTest, JwtPayloadAvailable) {
   EXPECT_TRUE(MessageDifferencer::Equals(expected_payload, *payload_));
 }
 
+TEST_F(ValidateJwtTest, OriginalPayloadOfExchangedToken) {
```

Comment:
Please add a test for the case when original payload was not there. Any may be another test with original payload, but jwt was not in the intended location.

Reply:
Done: added two more tests.